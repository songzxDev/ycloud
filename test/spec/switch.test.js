import ko from 'knockout'
import yswitch from '../../src/components/switch/index.js'
import $ from 'jquery'
describe('y-switch组件', () => {
  const PREFIX = 'y-'
  ko.components.register(PREFIX + yswitch.name, {
    viewModel: yswitch.init,
    template: yswitch.template,
    synchronous: true
  })
  ko.cleanNode(document.getElementsByTagName('body')[0])
  let vm = {
    switchChecked: ko.observable(false),
    switchDisabled: ko.observable(false)
  }
  document.body.innerHTML = ' <y-switch params="checked:switchChecked, disabled:switchDisabled "></y-switch>'
  ko.applyBindings(vm)
  var el = $('body')
  it("组件注册测试", () => {
    expect(yswitch).not.toBeNull();
    expect(ko.components.isRegistered(PREFIX + yswitch.name)).toBe(true);
  })
  it('生成switch测试', () => {
    expect($('body').find('.y-switch').length > 0).toEqual(true)
  })
  it('快照', () => {
    expect($('body').get(0)).toMatchSnapshot()
  })
})
