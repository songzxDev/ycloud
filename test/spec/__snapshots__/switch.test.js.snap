// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`y-switch组件 快照 1`] = `
<body>
   
  <y-switch
    params="checked:switchChecked, disabled:switchDisabled "
  >
    <div
      class="inline-block"
    >
      
  
      <span
        class="y-switch y-switch-"
        data-bind="css:sizeClass, click: checkSelect"
      >
        
    
        <input
          type="hidden"
          value="false"
        />
        
    
        <span
          class="y-switch-inner"
        >
          
          
          <!-- ko template: { nodes: $componentTemplateNodes } -->
          <!-- /ko -->
          
    
        </span>
        
  
      </span>
      

    </div>
    

  </y-switch>
</body>
`;
